modules/dnn/src/layers/elementwise_layers.cpp

```diff
@@ -813,20 +813,173 @@ struct BaseDefaultFunctor : public BaseFunctor
+    }
+
+    inline v_float32 v_erf(v_float32 v) {
+        v_float32 coef0 = vx_setall_f32(c_erf_coef0),
+                  coef1 = vx_setall_f32(c_erf_coef1),
+                  coef2 = vx_setall_f32(c_erf_coef2),
+                  coef3 = vx_setall_f32(c_erf_coef3),
+                  coef4 = vx_setall_f32(c_erf_coef4),
+                  coef5 = vx_setall_f32(c_erf_coef5);
+        v_float32 ones = vx_setall_f32(1.0f),
+                  neg_zeros = vx_setall_f32(-0.f),
+                  t = v_abs(v);
+        // sign(v)
+        v_float32 sign_mask = v_and(neg_zeros, v);
+
+        t = v_div(ones, v_fma(coef0, t, ones));
+        v_float32 r = v_fma(coef1, t, coef2);
+        r = v_fma(r, t, coef3);
+        r = v_fma(r, t, coef4);
+        r = v_fma(r, t, coef5);
+        // - v * v
+        v_float32 pow_2 = v_mul(v, v);
+        v_float32 neg_pow_2 = v_xor(neg_zeros, pow_2);
+        // - exp(- v * v)
+        v_float32 exp = v_exp(neg_pow_2);
+        v_float32 neg_exp = v_xor(neg_zeros, exp);
+        v_float32 res = v_mul(t, neg_exp);
+        res = v_fma(r, res, ones);
+        return v_xor(sign_mask, res);
+    }
+#endif
+}
 
-    explicit GeluFunctor() {}
+struct GeluFunctor : public BaseFunctor {
+    using Layer = GeluLayer;
+    int vlanes;
 
-    bool supportBackend(int backendId, int)
-    {
+    explicit GeluFunctor() {
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+    vlanes = VTraits<v_float32>::vlanes();
+#else
+    vlanes = 1;
+#endif
+    }
+
+    bool supportBackend(int backendId, int) {
         return backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_CUDA;
     }
 
-    inline float calculate(float x) const
-    {
-        return 0.5f * x * (1.0f + erf(x * M_SQRT1_2));
+    void apply(const float* srcptr, float* dstptr, int stripeStart, int len, size_t planeSize, int cn0, int cn1) const {
+        CV_UNUSED(stripeStart);
+        for (int cn = cn0; cn < cn1; cn++, srcptr += planeSize, dstptr += planeSize) {
+            int i = 0;
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+            // 0.5f * x * (1.0f + erf(x * M_SQRT1_2));
+            v_float32 half = vx_setall_f32(0.5f),
+                      one = vx_setall_f32(1.0f),
+                      reciprocal_sqrt2 = vx_setall_f32(M_SQRT1_2);
+            for (; i <= len - vlanes * 4; i += vlanes * 4) {
+                v_float32 x0 = vx_load(srcptr + i),
+                          x1 = vx_load(srcptr + i + vlanes),
+                          x2 = vx_load(srcptr + i + 2 * vlanes),
+                          x3 = vx_load(srcptr + i + 3 * vlanes);
+
+                // t = x * M_SQRT1_2
+                v_float32 t0 = v_mul(reciprocal_sqrt2, x0),
+                          t1 = v_mul(reciprocal_sqrt2, x1),
+                          t2 = v_mul(reciprocal_sqrt2, x2),
+                          t3 = v_mul(reciprocal_sqrt2, x3);
+
+                // t = 1.0f + t
+                t0 = v_add(one, pytorch_math::v_erf(t0));
```

Comment:
I suggest not to use pytorch_math namespace, because it may look confusing. It's enough to put the link to PyTorch code and use v_erf without any namespace (just use cv, cv::hal or cv::dnn)

Reply:
Ok, anonymous namespace is used. In this case, `erf` needs to be changed to `erf_approx` to avoid redef of the one from std library.